src/profiles_management/pmr/classes.py

```diff
@@ -0,0 +1,133 @@
+"""This package provides classes for validating and defining a PMR.
+
+The goal of these classes is to represent the PMR in a more
+Pythonic way, rather than a dict of vavlues.
+
+The classes should also provide some common helper methods for
+traversing the information of the PMR, but not functions for
+directly affecting the K8s cluster based on the PMR.
+
+This is similar to how the K8s python clients have classes
+for representing the objects and different methods for doing operations
+with them.
+"""
+
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List
+
+import jsonschema
+
+from profiles_management.pmr.schema import RESOURCE_QUOTA_SCHEMA
+
+log = logging.getLogger(__name__)
+
+
+class UserKind(Enum):
+    """Class representing the kind of the user."""
+
+    USER = "user"
+    SERVICE_ACCOUNT = "service-account"
+
+
+class ContributorRole(Enum):
+    """Class representing the role of the user."""
+
+    ADMIN = "admin"
+    EDIT = "edit"
+    VIEW = "view"
+
+
+@dataclass
+class Contributor:
+    """Class representing which users should have access to a Profile."""
+
+    name: str
+    role: ContributorRole
+
+
+@dataclass
+class Owner:
+    """Class representing the owner field of a Profile."""
+
+    name: str
+    kind: UserKind
+
+
+@dataclass
+class Profile:
+    """Class representing a Profile and its Contributors."""
+
+    name: str
+    owner: Owner
+    resources: Dict[str, Any]
+    contributors: List[Contributor]
+
+    # https://docs.python.org/3/library/dataclasses.html#post-init-processing
+    def __post_init__(self):
+        """Validate resourceQuota after values have been initialised."""
+        log.info("Validating ResourceQuota for Profile: %s", self.name)
+        jsonschema.validate(self.resources, RESOURCE_QUOTA_SCHEMA)
+        log.info("ResourceQuota is valid.")
+
+    def has_contributor(self, name: str, role: ContributorRole) -> bool:
+        """Check if the Profile has a contributor with specific role."""
+        for contributor in self.contributors:
+            if contributor.name != name:
+                continue
+
+            if contributor.role != role:
+                continue
+
+            log.info("Profile %s has contributor (%s, %s)", self.name, name, role.value)
+            return True
+
+        log.info("Profile %s doesn't have contributor (%s, %s)", self.name, name, role.value)
+        return False
+
+
+class ProfilesManagementRepresentation:
+    """A class representing the Profiles and Contributors."""
+
+    profiles: dict[str, Profile] = {}
+
+    def __init__(self, profiles: List[Profile] = []):
```

Comment:
Yep, should be resolved with `pydantic`. Will update shortly
https://github.com/canonical/github-profiles-automator/pull/12#discussion_r1877954448

Reply:
I updated the code to use Pydantic
https://github.com/canonical/github-profiles-automator/commit/630b04f0902b469508ee7fca4fa52dfbadeb02d6
https://github.com/canonical/github-profiles-automator/commit/f170187586cd83f125382b52dcfc859bef72f089

This should also handle the initialisation of fields, so contributors will be a `[]` by default. For the validation also of `profiles_list` I've used the [TypeAdapter](https://docs.pydantic.dev/latest/concepts/type_adapter/), since the PMR class doesn't inherit from `BaseModel`